replica/table.cc

```diff
@@ -3555,8 +3573,34 @@ future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locato
     }
 
     tlogger.info("Cleaned up tablet {} of table {}.{} successfully.", tid, _schema->ks_name(), _schema->cf_name());
+}
+
+future<> table::cleanup_tablet(database& db, db::system_keyspace& sys_ks, locator::tablet_id tid) {
+    auto holder = async_gate().hold();
+
+    auto* sg = storage_group_for_id(tid.value());
+
+    // Keep the storage group alive.
+    std::unique_ptr<storage_group> sg_ptr;
+    co_await _sg_manager->with_storage_groups_exclusive_lock([&] (storage_group_map& storage_groups) {
+        sg_ptr = std::move(storage_groups[tid.value()]);
+        storage_groups.erase(tid.value());
```

Comment:
While I understand that we don't want the tablet to be accessed after it's cleaned up, this creates an inconsistency between the tablet map and the storage_group map, I am not sure if this is the right place to deallocate the tablet. Ideally, or should it be done in update_effective_replication_map.

@tgrabiec maybe cleanup should happen as a side effect of update_effective_replication_map, prior to deallocating the storage_group. WDYT?